frontend-app/src/main/kotlin/no/nav/modialogin/utils/RedisUtils.kt

```diff
@@ -0,0 +1,38 @@
+package no.nav.modialogin.utils
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import no.nav.modialogin.Logging
+import no.nav.personoversikt.common.utils.EnvUtils.getRequiredConfig
+import redis.clients.jedis.Jedis
+import redis.clients.jedis.JedisPool
+
+object RedisUtils {
+    val pool: AuthJedisPool by lazy {
+        AuthJedisPool(
+            getRequiredConfig("REDIS_HOST"),
+            getRequiredConfig("REDIS_PASSWORD")
+        )
+    }
+}
+
+class AuthJedisPool(host: String, private val password: String) {
+    private val pool = JedisPool(host, 6379)
+    suspend fun <T> useResource(block: (Jedis) -> T): T? {
+        return withContext(Dispatchers.IO) {
+            if (pool.isClosed) {
+                Logging.log.error("JedisPool is closed while trying to access it")
+                null
+            } else {
+                runCatching {
+                    pool.resource.use {
+                        it.auth(password)
+                        block(it)
+                    }
+                }
+                    .onFailure { Logging.log.error("Redis-error", it) }
+                    .getOrNull()
+            }
+        }
+    }
+}
```

Comment:
Ettersom Redis instansen vår her alltid vil ha samme datatype for input her, kunne det vært bedre å flytte det generiske argumentet til klassen istedetfor funksjonen? Sånn at vi kun trenger å sende det inn en gang? Eller hva tenker du?

Reply:
`<T>` er bounded av retur-typen til `block`, så det er egentlig aldri noe vi sender inn.

f.eks
```kotlin
val value = redisPool.useResource { "This is a string"  } // `value` blir automatisk til "String"
val other = redisPool.useResource { 1231  }                   // `other` blir automatisk til "Int"
```

Flytter man `<T>` til klasse nivå ville koden ovenfor heller ikke fungert, siden `redisPool` ville vært "låst" til å alltid returnere samme type fra `useResource`. Så da ville man fått trøbbel fordi `redis.get(key)` returnerer `String?` mens `redis.del(key)` returnerer `Long`